src/createRestyleComponent.tsx

```diff
@@ -6,7 +6,7 @@ import useRestyle from './hooks/useRestyle';
 
 const createRestyleComponent = <
   Props extends Record<string, any>,
-  Theme extends BaseTheme = BaseTheme
+  Theme extends BaseTheme
```

Comment:
Removing the default since if neither props or Theme is provided than it can be inferred, but if props is provided then we should require user to also pass in Theme so that the restyle functions passed in can be checked for proper typesafety